services/frontend/adapters/inbound/controller_validator.go

```diff
@@ -0,0 +1,66 @@
+package inboundAdapters
+
+import (
+	"context"
+	"fmt"
+
+	"gopkg.in/yaml.v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/berops/claudie/internal/manifest"
+)
+
+// secretValidator validates Secrets
+type secretValidator struct{}
```

Comment:
I think it would make more sense to have this under the domain, and reference it from `usecases` struct.

i.e. in `controller.go`

```go
hookServer.Register(mc.validationWebhookPath, admission.WithCustomValidator(&corev1.Secret{}, usecases.SecretValidator))
```